src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationClaimantOfHearingHandler.java

```diff
@@ -72,31 +65,67 @@ public List<CaseEvent> handledEvents() {
+        if (!isApplicantLip(caseData)) {
+            if (caseData.getSolicitorReferences() == null
+                || caseData.getSolicitorReferences().getApplicantSolicitor1Reference() == null) {
+                reference = "";
+            } else {
+                reference = caseData.getSolicitorReferences().getApplicantSolicitor1Reference();
+            }
+            return new HashMap<>(Map.of(
+                CLAIM_REFERENCE_NUMBER,
+                legacyCaseRef,
+                HEARING_FEE,
+                caseData.getHearingFee() == null ? "£0.00" : String.valueOf(caseData.getHearingFee().formData()),
+                HEARING_DATE,
+                hearingDate,
+                HEARING_TIME,
+                hearingTime,
+                HEARING_DUE_DATE,
+                caseData.getHearingDueDate() == null ? "" :
+                    caseData.getHearingDueDate().format(DateTimeFormatter.ofPattern("dd-MM-yyyy")),
+                CLAIMANT_REFERENCE_NUMBER, reference
+            ));
         } else {
-            reference = caseData.getSolicitorReferences().getApplicantSolicitor1Reference();
+            return new HashMap<>(Map.of(
+                CLAIM_REFERENCE_NUMBER,
+                legacyCaseRef,
+                HEARING_DATE,
+                hearingDate,
+                HEARING_TIME,
+                hearingTime
+            ));
         }
-        return new HashMap<>(Map.of(
-            CLAIM_REFERENCE_NUMBER,
-            caseData.getLegacyCaseReference(),
-            HEARING_FEE,
-            caseData.getHearingFee() == null ? "£0.00" : String.valueOf(caseData.getHearingFee().formData()),
-            HEARING_DATE,
-            caseData.getHearingDate().format(DateTimeFormatter.ofPattern("dd-MM-yyyy")),
-            HEARING_TIME,
-            time.format(DateTimeFormatter.ofPattern("hh:mma")).replace("AM", "am").replace("PM", "pm"),
-            HEARING_DUE_DATE,
-            caseData.getHearingDueDate() == null ? "" :
-                caseData.getHearingDueDate().format(DateTimeFormatter.ofPattern("dd-MM-yyyy")),
-            CLAIMANT_REFERENCE_NUMBER, reference
+    }
+
+    private boolean isApplicantLip(CaseData caseData) {
+        return (YesOrNo.NO.equals(caseData.getApplicant1Represented()));
+    }
+
+    private String getRecipient(CaseData caseData, boolean isApplicantLip) {
+        return isApplicantLip ? caseData.getApplicant1().getPartyEmail()
+            : caseData.getApplicantSolicitor1UserDetails().getEmail();
+    }
 
-        ));
+    private String getReferenceTemplate(CaseData caseData, boolean isApplicantLip) {
+        return isApplicantLip ? String.format(REFERENCE_TEMPLATE_HEARING_LIP, caseData.getHearingReferenceNumber())
+            : String.format(REFERENCE_TEMPLATE_HEARING, caseData.getHearingReferenceNumber());
+    }
+
+    private String getEmailTemplate(CaseData caseData, boolean isApplicantLip) {
```

Comment:
we can rewrite in this way.

private String getEmailTemplate(CaseData caseData, boolean isApplicantLip) {
    // Check if applicant is not a LIP (Litigant in Person)
    if (!isApplicantLip) {
        // Check if hearing fee is not null and greater than zero
        if (caseData.getHearingFee() != null && caseData.getHearingFee().getCalculatedAmountInPence().compareTo(BigDecimal.ZERO) > 0) {
            return notificationsProperties.getHearingListedFeeClaimantLrTemplate();
        } else {
            return notificationsProperties.getHearingListedNoFeeClaimantLrTemplate();
        }
    }
    // Applicant is a LIP, return the corresponding template
    return notificationsProperties.getHearingNotificationLipDefendantTemplate();
}

what do you think?

Reply:
Don't quite get that, but I'll refactor it in a better way.